androidApp/src/main/java/com/example/myplants/android/MainActivity.kt

```diff
@@ -3,29 +3,95 @@ package com.example.myplants.android
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.material.*
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.lifecycleScope
 import com.example.myplants.Greeting
+import com.example.myplants.plants.domain.Plant
+import com.example.myplants.plants.domain.PlantRepository
+import com.example.myplants.plants.domain.PlantSize
+import kotlinx.coroutines.launch
+import kotlinx.datetime.DayOfWeek
+import kotlinx.datetime.toKotlinLocalTime
+import org.koin.android.ext.android.inject
+import java.util.UUID
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        val plantRepository: PlantRepository by inject()
+
+        var plantsList = emptyList<Plant>()
+        /*
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                plantRepository.getPlants().collectLatest {
+                    plantsList = it
+                }
+            }
+        }
+         */
         setContent {
             MyApplicationTheme {
                 Surface(
                     modifier = Modifier.fillMaxSize(),
                     color = MaterialTheme.colors.background
                 ) {
                     GreetingView(Greeting().greet())
+                    PlantName(onClick = { plant ->
+                        lifecycleScope.launch {
+                            plantRepository.upsertPlant(plant)
+                        }
```

Comment:
In Compose, we use a coroutine scope which is aware of the current composables lifecycle:
```
val scope = rememberCoroutineScope()
Button(onClick = { scope.launch { ... } } )
```